hudi-client/hudi-spark-client/src/main/java/org/apache/hudi/commmon/model/HoodieSparkRecord.java

```diff
@@ -47,42 +50,49 @@
 import java.util.Properties;
 
 import static org.apache.hudi.common.table.HoodieTableConfig.POPULATE_META_FIELDS;
+import static org.apache.hudi.util.HoodieSparkRecordUtils.getNullableValAsString;
+import static org.apache.hudi.util.HoodieSparkRecordUtils.getValue;
 import static org.apache.spark.sql.types.DataTypes.BooleanType;
 import static org.apache.spark.sql.types.DataTypes.StringType;
 
 /**
  * Spark Engine-specific Implementations of `HoodieRecord`.
  */
+@DefaultSerializer(HoodieSparkRecordSerializer.class)
 public class HoodieSparkRecord extends HoodieRecord<InternalRow> {
```

Comment:
We need to make sure that 

 - We only hold `UnsafeRow` w/in `HoodieSparkRecord` (for the same reasons we hold bytes in `HoodieLegacyAvroRecord`); whenever we instantiate HoodieSparkRecord we need to check whether passed in InternalRow is UnsafeRow and if not we should do UnsafeProjection (that will serialize it to bytes)

Reply:
The HoodieSparkRecord only be instantiated by SparkParquetReader or instantiated by DataFrame in HoodieSparkSqlWriter. We can make sure that HoodieSparkRecord only hold UnsafeRow now.

Reply:
@wzx140 this has to be an invariant -- `HoodieSparkRecord` have to only hold `UnsafeRow` (as i called out before for the same reasons HoodieRecordPayload holds bytes and not objects). 

This is necessary to make sure

 - We don't hold in memory a lot of small objects (for every cell in every row)
 - De/Serialization procedure is simple (we write bytes in, we read bytes out, no object deserialization is performed)

Reply:
Fixed